docs/bots-guide.md

```diff
@@ -123,15 +123,7 @@ You need:
       explained [here](#configuration-file)).
     * Copy the file to a destination of your choice, e.g. to `~/.zuliprc`.
 
-4. Subscribe the bot to the streams that the bot needs to interact with.
-
-    * To subscribe your bot to streams, navigate to *Manage
```

Comment:
Wait, wait, I think subscription is sometimes necessary for the case when the bot is parsing for each message in the stream and do something when the message contains a keyword, for instance. The realm-to-realm mirror bot is another example.

Reply:
Hmm, good thought! So for the zulip bot framework, we have limited bots to react only to messages where they get @-mentioned. 

Maybe we want to add a new section for special bots like realm-to-realm bots, where the default limitations don't apply?

Reply:
I think we want to document that somewhere else.  Basically, if a bot is a standard bot that interacts with PMs/mentions, it doesn't need to be subscribed directly.  If it wants all messages on a stream, it needs to be subscribed.  Probably the documentation for the individual bot should cover that if needed for that bot (we can make this one of our docs macros to avoid duplicating text).  

And then maybe we'll also want to discuss subscribing to streams somewhere in the documentation on writing bots, as an option ("If your bot needs to see all messages on a stream, not just those where it is mentioned...").